### PR TITLE
OfferCondition model improvements

### PR DESCRIPTION
**Pull request by despo**
## Context

- Move STANDARD_CONDITIONS to OfferCondition
- Enable OfferCondition to check if it's part of standard conditions. This should enable us to simplify checks run by other services.

## Link to Trello card

https://trello.com/c/ZbXOjpKS/3827-refactor-offercondition-model-improvements

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Review by benswannack — COMMENTED** [review]
This is going to have more of an impact once the two form PRs are merged. Let's hold off on this for now (or rebase on the support form branch)

**Review by cwrw — APPROVED** [review]

**Pull request merged**
